File: accidental_sharing/accidental_sharing.py
class Person:
    def __init__(self, foods):
        self.favorite_foods = foods

donuts = ["donuts"]

alice = Person(donuts)
alice.favorite_foods.append("apples")

bob = Person(donuts)
bob.favorite_foods.append("bananas")

print("alice:", alice.favorite_foods);
print("  bob:", bob.favorite_foods);

File: accidental_sharing/accidental_sharing.rs
struct Person {
    favorite_foods: Vec<&'static str>,
}

fn main() {
    let donuts = vec!["donuts"];

    let mut alice = Person {
        favorite_foods: donuts, // value moved here
    };
    alice.favorite_foods.push("apples");

    let mut bob = Person {
        favorite_foods: donuts, // error: use of moved value: `donuts`
    };
    bob.favorite_foods.push("bananas");

    println!("{:?}", alice.favorite_foods);
    println!("{:?}", bob.favorite_foods);
}
